crates/ty_python_semantic/src/types/call/bind.rs

```diff
@@ -596,6 +598,48 @@ impl<'db> Bindings<'db> {
                         }
                     }
 
+                    function @ Type::FunctionLiteral(function_type)
+                        if dataclass_field_specifiers.contains(&function)
```

Comment:
Hm, this is probably too naive for overloaded functions (which is common for field specifiers)

Reply:
... no, that's fine